File: ScrollView/app/src/main/java/com/hfad/scrollview/MainActivity.java
package com.hfad.scrollview;

import androidx.appcompat.app.AppCompatActivity;

import android.os.Bundle;
import android.widget.TextView;

public class MainActivity extends AppCompatActivity {

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_main);
        TextView mTittleWindows = (TextView) findViewById(R.id.titleWindow);
        TextView mMessageWindows = (TextView) findViewById(R.id.messageWindow);

        mTittleWindows.setText("This is some Title.");

        StringBuilder stringBuilder = new StringBuilder();

        String someMessage = " This is some message that I'm writtiing to appand. ";
        for (int i =0; i < 100; i++) {
            stringBuilder.append(someMessage);
        }

        mMessageWindows.setText(stringBuilder.toString());

    }
}

File: KotlinCountDownTimer/app/src/main/java/com/hfad/kotlincountdowntimer/MainActivity.kt
package com.hfad.kotlincountdowntimer

import androidx.appcompat.app.AppCompatActivity
import android.os.Bundle
import android.os.CountDownTimer
import com.hfad.kotlincountdowntimer.databinding.ActivityMainBinding

class MainActivity : AppCompatActivity() {
    private lateinit var binding: ActivityMainBinding
    private var timer: CountDownTimer? = null
    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        binding = ActivityMainBinding.inflate(layoutInflater)
        setContentView(binding.root)
        binding.apply {
            btnStart.setOnClickListener {

                startCountDownTimer(20000)
            }
        }
    }

    private fun startCountDownTimer(timerMillis: Long) {
        timer?.cancel()
        timer = object : CountDownTimer(timerMillis, 1) {
            override fun onTick(timeM: Long) {
                binding.tvTimer.text = timeM.toString()
            }

            override fun onFinish() {
                binding.tvTimer.text = "Finish"
            }

        }.start()
    }

}